core-processor/src/main/java/io/micronaut/inject/ast/ClassElement.java

```diff
@@ -121,6 +122,30 @@ default boolean isOptional() {
         return isAssignable(Optional.class) || isAssignable(OptionalLong.class) || isAssignable(OptionalDouble.class) || isAssignable(OptionalInt.class);
     }
 
+    /**
+     * Checks whether the bean type is a container type.
+     * @return Whether the type is a container type like {@link Iterable}.
+     * @since 4.0.0
+     */
+    default boolean isContainerType() {
```

Comment:
Where do you suggest?

Reply:
I moved `isProvider` out since there was only one use just to a private method. The `isContainerType` method doesn't seem to have to do with DI as it relates to collections so I left it